.github/workflows/ci-manual-build.yaml

```diff
@@ -0,0 +1,51 @@
+name: "[CI] Build"
+on:
+  workflow_call:
+    inputs:
+      git_sha:
+        required: true
+        type: string
+      build_id:
+        required: true
+        type: string
+      goal:
+        required: true
+        type: choice
+        default: "test"
+        options:
+          - "build"
+          - "launch"
+          - "test"
+      k8s_version:
+        required: true
+        type: string
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.git_sha }}
+      - uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: ${{ secrets.AWS_REGION }}
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+      - run: echo "AMI_NAME=amazon-eks-node-${{ inputs.k8s_version }}-${{ inputs.build_id }}" >> $GITHUB_ENV
+      - run: make ${{ inputs.k8s_version }} ami_name=${{ env.AMI_NAME }}
+      - id: ami_id
+        run: echo "ami_id=$(jq -r .builds[0].artifact_id "${{ env.AMI_NAME }}-manifest.json" | cut -d ':' -f 2)" >> $GITHUB_OUTPUT
+    outputs:
+      ami_id: ${{ steps.ami_id.outputs.ami_id }}
+  integration-test:
```

Comment:
i want call it odd for `build` to have `integration-test` as a step, versus having one workflow or calling both separately but i guess this makes [passing the k8s version matrix simpler for the caller](https://github.com/awslabs/amazon-eks-ami/pull/1402/files#diff-9d2ab28a7718195cd81484529ba18bb65e4f012aad6a2f57204be4472b60ac9eR69)

Reply:
It's definitely odd, and (having written this workflow a while ago) I don't really remember what limitation made this the way to go. Let me see if I can fix that...

Reply:
The nesting of workflows shows up in the UI, but other than that, not much benefit. I refactored these as actions in `.github/actions/ci/`, PTAL @ndbaker1

Reply:
looks awesome 🎉, neat expanding coverage of actions depending on input `goal`